### ISSUE TITLE
Regarding the issue where --mask-mode segment cannot be computed on an RTX 4090 machine.

### ISSUE THREAD

**Issue by StevenssW** [MrNeRF/LichtFeld-Studio#577]
I tried processing data with masks. When I run the same version of the code on an RTX 4070 machine, it works normally. However, when I move the computation to an RTX 4090 machine, the process gets stuck, and both the CPU and GPU remain idle. I tested this on two different RTX 4090 machines and encountered the same issue, so I’m not sure what the cause is.

When I delete the masks folder, the computation runs successfully on the RTX 4090 as well. However, if I do not use --mask-mode segment and instead generate the masks folder manually, the RTX 4090 still fails to run.

The computation stops at this point — it neither terminates nor continues :

`(lichtfeld-studio) root@04aa96fce798:/ailab/LichtFeld-Studio# ./build/LichtFeld-Studio -d /ailab/data/sfm -o  /ailab/data/output/sfm  --mask-mode segment  --strategy mcmc --max-cap 1000000 --headless  --iter 25000

Parameter verification report:

Mismatched values:
  - means_lr: JSON=1.6e-05, Default=0.00016 (Initial learning rate for position updates)

Parameters in struct but not in JSON:
  - final_lr_fraction (Final LR as fraction of initial (0.01 = decay to 1%))
  - num_workers (Number of image loader threads)
  - gpu_id (GPU device ID to use (-1 = auto/default, 0+ = specific GPU))
  - bilateral_grid_warmup_steps (Warmup steps for bilateral grid scheduler)
  - bilateral_grid_warmup_start_lr (Starting LR factor for bilateral grid warmup)
  - enable_sparsity (Enable sparsity optimization)
  - sparsify_steps (Number of steps for sparsification)
  - init_rho (Initial ADMM penalty parameter)
  - prune_ratio (Final pruning ratio for sparsity)
  - save_sog (Save in SOG format alongside PLY)
  - sog_iterations (K-means iterations for SOG compression)

Unknown parameters in JSON (will be ignored):
  - bg_modulation
  - gut
[20:35:17.715] [info] main.cpp:40  ========================================
[20:35:17.715] [info] main.cpp:41  LichtFeld Studio
[20:35:17.715] [info] main.cpp:42  ========================================
[20:35:17.715] [info] application.cpp:23  Starting headless training...
[20:35:17.716] [info] project.cpp:636  project_path is empty - creating new project.lfs file
[20:35:17.726] [info] training_setup.cpp:29  Loading dataset from: /ailab/data/sfm
[20:35:17.734] [info] loader_service.cpp:66  Using COLMAP loader for: /ailab/data/sfm
[20:35:17.775] [info] colmap_loader.cpp:70  Loading COLMAP in binary format
[20:35:17.898] [info] colmap.cpp:943  Read 1 cameras and 103 images from COLMAP
[20:35:18.081] [info] colmap.cpp:894  Training with 103 images
Dataset created with 103 images (split: 2)
[20:35:18.770] [info] colmap_loader.cpp:226  Loaded 14693 points from COLMAP
[20:35:18.770] [info] colmap_loader.cpp:255  COLMAP dataset loaded successfully in 1035ms
[20:35:18.770] [info] colmap_loader.cpp:256    - 103 cameras
[20:35:18.770] [info] training_setup.cpp:35  Dataset loaded successfully using COLMAP loader
[20:35:18.770] [info] training_setup.cpp:77  Using point cloud with 14693 points
Scene scale: 0.61011344
Initialized SplatData with:
  - 14693 points
  - Max SH degree: 3
  - Total SH coefficients: 16
  - sh0 shape: [14693, 1, 3]
  - shN shape: [14693, 15, 3]
[20:35:18.879] [info] trainer.cpp:487  Initializing trainer with 25000 iterations
[20:35:18.879] [info] trainer.cpp:511  Using default GPU 0 (auto mode)
[20:35:18.879] [info] trainer.cpp:532  Using all 103 images for training (no evaluation)
[20:35:18.879] [info] trainer.cpp:79  Checking masks for 103 cameras...
[20:35:18.987] [info] trainer.cpp:98  Found 103 mask files
[20:35:18.989] [info] trainer.cpp:645  Render mode: RGB
[20:35:18.989] [info] trainer.cpp:646  Visualization: disabled
[20:35:18.989] [info] trainer.cpp:647  Strategy: mcmc
[20:35:18.989] [info] trainer.cpp:650  Trainer initialization complete
[20:35:18.989] [info] trainer.cpp:1139  Starting training loop with 16 workers
[20:35:19.053] [info] cache_image_loader.cpp:453  all images can be cached in RAM. Cache Loader will be in CPU memory mode.`

**Comment by MrNeRF** [comment]
It is hard for me to get any information from your description which could hint to what is actually going on. 
Which commit are you on? Have you tested latest commit on master?

**Comment by StevenssW** [comment]
> It is hard for me to get any information from your description which could hint to what is actually going on. Which commit are you on? Have you tested latest commit on master?

Thanks！ I updated to the latest code and recompiled it, and it no longer reports errors. However, after the computation finishes, it prints Segmentation fault (core dumped).

In addition, I noticed that a Docker image compiled on an AMD CPU cannot run properly on an Intel CPU. Could this be related to the -march=native compilation flag?

`Training [██████████████████████████████████████▌░] 97% [05m:08s<00m:09s] 24300/25000 | Loss: 0.0325 | Splats: 1000000 (Training [███████████████████████████████████████▌] 98% [05m:09s<00m:06s] 24400/25000 | Loss: 0.0307 | Splats: 1000000 (Training [███████████████████████████████████████▌] 98% [05m:11s<00m:06s] 24500/25000 | Loss: 0.0263 | Splats: 1000000 (Training [███████████████████████████████████████▌] 98% [05m:12s<00m:06s] 24600/25000 | Loss: 0.0280 | Splats: 1000000 (Training [███████████████████████████████████████▌] 99% [05m:14s<00m:03s] 24700/25000 | Loss: 0.0426 | Splats: 1000000 (Training [███████████████████████████████████████▌] 99% [05m:15s<00m:03s] 24800/25000 | Loss: 0.0508 | Splats: 1000000 (Training [████████████████████████████████████████] 100% [05m:17s<00m:00s] 24900/25000 | Loss: 0.0400 | Splats: 1000000 (+)
[21:38:21.874] [info] ply.cpp:865  PLY saved: /ailab/data/output/sfm/splat_25000.ply
[21:38:24.566] [info] checkpoint.cpp:72  Checkpoint saved: /ailab/data/output/sfm/checkpoints/checkpoint_25000.resume (1000000 Gaussians, iter 25000)

✓ Training completed in 335.565s (avg 74.5 iter/s)
✓ Final splats: 1000000
[21:38:24.595] [info] image_io.cpp:544  [BatchImageSaver] Starting with 4 worker threads
[21:38:24.596] [info] trainer.cpp:1281  Training completed successfully
[21:38:24.600] [info] application.cpp:58  Headless training completed
[21:38:24.611] [warn] memory_arena.cu:360  Emergency cleanup started
[21:38:24.617] [warn] memory_arena.cu:388  Emergency cleanup completed
[21:38:24.638] [info] image_io.cpp:558  [BatchImageSaver] Shutting down...
[21:38:24.638] [info] image_io.cpp:570  [BatchImageSaver] Shutdown complete
[21:38:24.639] [info] memory_arena.cu:982  Arena stats: committed=786 MB, peak=1408 MB (179.3%), frames=25000, reallocs=17, runtime=335s
Segmentation fault (core dumped)`

**Comment by kmeirlaen** [comment]
Closed as original request was resolved - please open a new issue for other topics if still applicable.

**Comment by qianqiuzzz** [comment]
@StevenssW did you solve the problem? i met  the same
